crates/goose-bench/src/eval_suites/vibes/flappy_bird.rs

```diff
@@ -0,0 +1,121 @@
+use crate::bench_work_dir::BenchmarkWorkDir;
+use crate::eval_suites::{
+    collect_baseline_metrics, copy_session_to_cwd, metrics_hashmap_to_vec, BenchAgent, Evaluation,
+    EvaluationMetric, ExtensionRequirements,
+};
+use crate::register_evaluation;
+use async_trait::async_trait;
+use goose::message::MessageContent;
+use mcp_core::role::Role;
+use serde_json::{self, Value};
+use std::fs;
+
+pub struct FlappyBird {}
+
+impl FlappyBird {
+    pub fn new() -> Self {
+        FlappyBird {}
+    }
+
+    fn check_python_implementation(&self, content: &str) -> bool {
+        content.contains("import pygame") &&
+        content.contains("pygame.init()") &&
+        content.contains("while") && // Game loop
+        content.contains("pygame.event.get()") && // Event handling
+        content.contains("def main") && // Main function
+        content.contains("if __name__ == '__main__'") // Main guard
+    }
+}
+
+#[async_trait]
+impl Evaluation for FlappyBird {
+    async fn run(
+        &self,
+        mut agent: Box<dyn BenchAgent>,
+        work_dir: &mut BenchmarkWorkDir,
+    ) -> anyhow::Result<Vec<(String, EvaluationMetric)>> {
+        println!("FlappyBird - run");
+
+        // Collect baseline metrics (execution time, token usage, tool calls)
+        let (messages, perf_metrics) = collect_baseline_metrics(
+            &mut agent,
+            "Create a Flappy Bird game in Python. Structure the code with a main function and use the if __name__ == '__main__': idiom. You must use pygame. The background color should be a light blue color. Pressing SPACE multiple times will accelerate the bird. The bird's shape should be a red circle. Place on the bottom some land colored as dark yellow chosen. Make a score shown on the top right side. Increment if you pass pipes and don't hit them. Make randomly spaced dark green pipes with enough space. When you lose, show the best score. Make the text inside the screen. Pressing q or Esc will quit the game. Restarting is pressing SPACE again. The final game should be written to a file named flappy_bird.py. Remember to use your tools if applicable.".to_string()
```

Comment:
maybe have it create a python virtual env? mine installed pygame using the current python (which I didn't care about) but I could imagine this maybe messing up environments?